vector/v.sample/main.c

```diff
@@ -215,7 +215,7 @@ int main(int argc, char **argv)
         type = Vect_read_line(&In, Points, Cats, line);
         if (!(type & GV_POINT))
             continue;
-        if (field != -1 && !Vect_cat_get(Cats, field, &cat))
+        if (field != -1 && !Vect_cat_get(Cats, field, &cat)) // 'cat' is used uninitialized whenever '&&' condition is false
```

Comment:
Here, it's actually never used in that case, right? Just making sure cat = 0 (or the previous undefined) does not leak anywhere. (Maybe similar question will apply to other parts, too.)

Reply:
[Vect_cat_get()](https://grass.osgeo.org/programming8/vector_2Vlib_2cats_8c.html#a3d9d5db7d953e0a1b36b63e19a91b903j) will set `cat`:

> to first category found or -1 if field was not found

so it doesn’t matter what value we initialize it to (but I’ll change this one to `-1`). The compiler doesn’t know this.

Re the other cases: there are quite a few with if..else or similar that covers all options. Again, the compiler doesn’t know this.